test/functionalTest/cases/3154_notification_problem_with_custom_header/notification_problem_with_custom_header.test

```diff
@@ -0,0 +1,204 @@
+# Copyright 2018 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Notification Templates, with refresh of the subscription cache
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 0
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create custom subscription
+# 02. Create entity to trigger the subscription
+# 03. Dump accumulator to see the notification
+# 04. GET the subscription
+#
+
+echo "01. Create custom subscription"
+echo "=============================="
+payload='{
+  "description": "x_yz4hahh",
+  "expires": "2040-01-01T14:00:00.00Z",                  
+  "subject": {
+    "entities": [
+      {
+        "idPattern" : ".*"
+      }
+    ],
+    "condition": {
+      "attrs": []
+    }
+  },
+  "notification": {
+    "attrs": [],
+    "attrsFormat": "legacy",
+    "httpCustom": {
+      "url":      "http://localhost:'${LISTENER_PORT}'/notify"
```

Comment:
What about if "attrsFormat": "legacy" and a custom "payload" are used at the same time? My guess is that "payload" will take precedence over attrsFormat, but it would be good to check it with a new .test.

Reply:
New test case in functest in 4b8531622